edgelet/kube-client/src/client.rs

```diff
@@ -13,14 +13,15 @@ use hyper_tls::HttpsConnector;
 use k8s_openapi::api::apps::v1 as apps;
 use k8s_openapi::api::authentication::v1 as auth;
 use k8s_openapi::api::core::v1 as api_core;
+use k8s_openapi::api::rbac::v1 as rbac;
 use k8s_openapi::apimachinery::pkg::apis::meta::v1 as api_meta;
 use k8s_openapi::{http, Response as K8sResponse, ResponseBody};
 use log::debug;
 
 use crate::config::{Config, TokenSource};
 use crate::error::{Error, ErrorKind};
 
-pub struct HttpClient<C, B>(HyperClient<C, B>);
+pub struct HttpClient<C, B>(pub HyperClient<C, B>);
```

Comment:
Curious, why did this need to be made public?

Reply:
If I want to create an instance of `HttpClient` outside `edgelet-kube` crate I cannot do this because `HyperClient<C,B>` is private. I know 3 alternative way to do so:
1. use `pub` like I did here
2. add `new` method to `impl` block
3. do not use `HttpClient` at all by adding another struct where required that implements `Service` trait

Reply:
Sure, but why did we need to directly create an `HttpClient` from outside the `kube-client` crate though? Shouldn't this be created indirectly when we call `Client::new`?

Reply:
`Client::new()` creates HTTPS client whereas I need HTTP client in my tests in order to leverage `run_tcp_server()` functionality you added.

Reply:
Isn't there a `with_client` function you could use for that?

Reply:
Decided to live it like this so far.